src/metabase/domain_entities/specs.clj

```diff
@@ -0,0 +1,90 @@
+(ns metabase.domain-entities.specs
+  (:require [medley.core :as m]
+            [metabase.mbql
+             [normalize :as mbql.normalize]
+             [util :as mbql.u]]
+            [metabase.util.yaml :as yaml]
+            [schema
+             [coerce :as sc]
+             [core :as s]]))
+
+(def MBQL
+  "MBQL clause (ie. a vector starting with a keyword)"
+  (s/pred mbql.u/mbql-clause?))
+
+(def FieldType
+  "Field type designator -- a keyword derived from `type/*`"
+  (s/constrained s/Keyword
+                                        ;#(isa? % :type/*)
+                 identity))
+
+(def ^:private DomainEntityReference s/Str)
+
+(def ^:private DomainEntityType (s/constrained s/Keyword #(isa? % :DomainEntity/*)))
```

Comment:
There's actually an `s/isa` schema you can use here instead which will probably give you better error messages if the check fails

Reply:
bump

Reply:
stale